libraries/arrow-convert/src/from_impls.rs

```diff
@@ -136,6 +148,18 @@ impl<'a> TryFrom<&'a ArrowData> for &'a [u8] {
     }
 }
 
+impl<'a> TryFrom<&'a ArrowData> for Vec<u8> {
+    type Error = eyre::Report;
+    fn try_from(value: &'a ArrowData) -> Result<Self, Self::Error> {
+        let array: &PrimitiveArray<arrow::datatypes::UInt8Type> =
+            value.as_primitive_opt().wrap_err("not a primitive array")?;
+        if array.null_count() != 0 {
+            eyre::bail!("array has nulls");
+        }
+        Ok(array.values().to_vec())
```

Comment:
Here we could reuse the `TryFrom` implementation for `&[u8]` to reduce code duplication.

Reply:
Indeed!